File: services/scanner/internal/handlers/health.go
// Package handlers - HTTP request handlers
package handlers

import (
	"net/http"
	"time"

	"github.com/gin-gonic/gin"
)

var startTime = time.Now()

// HealthCheck returns the health status of the service
func HealthCheck(c *gin.Context) {
	c.JSON(http.StatusOK, gin.H{
		"status":    "healthy",
		"version":   "2.0.0",
		"service":   "scanner",
		"timestamp": time.Now().Format(time.RFC3339),
		"uptime":    time.Since(startTime).String(),
	})
}

// ReadyCheck returns whether the service is ready to accept requests
func ReadyCheck(c *gin.Context) {
	c.JSON(http.StatusOK, gin.H{
		"ready": true,
	})
}

// LiveCheck returns whether the service is alive
func LiveCheck(c *gin.Context) {
	c.JSON(http.StatusOK, gin.H{
		"live": true,
	})
}

File: services/ai/app/routes/health.py
"""
Health check routes
"""

from fastapi import APIRouter
from datetime import datetime
import time

router = APIRouter()
start_time = time.time()


@router.get("/health")
async def health_check():
    """Health check endpoint."""
    return {
        "status": "healthy",
        "version": "2.0.0",
        "service": "ai",
        "timestamp": datetime.now().isoformat(),
        "uptime_seconds": round(time.time() - start_time, 2)
    }


@router.get("/health/ready")
async def ready_check():
    """Readiness probe for Kubernetes."""
    return {"ready": True}


@router.get("/health/live")
async def live_check():
    """Liveness probe for Kubernetes."""
    return {"live": True}
